src/lib.rs

```diff
@@ -551,10 +576,9 @@ impl<S: SuffixScheme> FileRotate<S> {
             new_suffix_info
         };
 
-        let old_path = match old_suffix_info {
-            Some(suffix) => suffix.to_path(&self.basepath),
-            None => self.basepath.clone(),
-        };
+        if new_path.exists() {
+            return Err(MoveFileError::SuffixScanNeeded);
+        }
```

Comment:
This seems very racy to me. If the `new_path.exists() == false`, a user can still modify the filesystem before we reach `fs::rename` on line 586. What is the intent of the code here? Won't rescanning suffixes just overwrite said file anyway?

Reply:
You're right, I'll fix the race condition.
Rescanning suffixes does not overwrite anything in the fs, it just recreates `self.suffixes` which is only an internal cache of which log files are on disk. For correct functioning, we rely on the assumption that `self.suffixes` is correct. If it is not correct (user/system has done file ops), and if not for the asserts which up until now have served as the sole line of defense, we risk overwriting previous log files. Because `fn move_file_with_suffix` will only move away the destination file if it thinks it exists (look into the recursive nature of the function - it calls itself before `fs::rename` for this purpose).